cube/src/main/java/zone/cogni/asquare/cube/index/IndexFolderService.java

```diff
@@ -96,11 +96,14 @@ private List<String> getLocalPaths(@Nonnull Resource[] resources) {
   }
 
   private String getLocalPathPrefix() {
+    String result = null;
     if (configurationPath.startsWith("classpath:"))
-      return StringUtils.substringAfter(configurationPath, "classpath:");
+      result = StringUtils.substringAfter(configurationPath, "classpath:");
     else if (configurationPath.startsWith("file:"))
-      return StringUtils.substringAfter(configurationPath, "file:");
-
+      result = StringUtils.substringAfter(configurationPath, "file:");
+    if (result != null) {
```

Comment:
this reminds me: normal flow should be free of if tests. exceptions should be in if-tests typically.

so that would mean more like if (result == null) throw exception.